code/game/gamemodes/gang/gang.dm

```diff
@@ -248,6 +249,19 @@ GLOBAL_LIST_INIT(gang_colors_pool, list("red","orange","yellow","green","blue","
 		gang_bosses += G.bosses
 	return gang_bosses
 
+/datum/game_mode/proc/shuttle_check()
+	if(forced_shuttle)
+		return
+	var/alive = 0
+	for(var/mob/living/L in GLOB.player_list)
+		if(L.stat != DEAD)
+			alive++
+	if(alive < LAZYLEN(GLOB.joined_player_list) * 0.4)
+		priority_announce("Catastrophic casualties detected: crisis shuttle protocols activated - jamming recall signals across all frequencies.")
+		forced_shuttle = TRUE
+		if(SSshuttle.emergency.timeLeft(1) < (SSshuttle.emergencyCallTime * 0.4))
```

Comment:
The shuttle can get recalled up to `emergencyCallTime * 0.25` aka 2.5 minutes during code red or delta; since both are likely in gang this shuttle could feasibly be recalled and then because of the `forced_shuttle` check this proc would never execute again.
https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/shuttle.dm#L248